File: backend/src/main/java/info/gamewise/lor/videos/deckcodeextractor/DescriptionWithDeckLinkExtractor.java
package info.gamewise.lor.videos.deckcodeextractor;

import java.util.Optional;

public class DescriptionWithDeckLinkExtractor extends AbstractDeckCodeExtractor implements DeckCodeExtractor {

    @Override
    public Optional<String> extract(String videoDescription) {
        return descriptionAsStream(videoDescription)
                .filter(this::filterMobalyticsLink)
                .findFirst()
                .map(this::deckCodeFromMobaLinkUrl);
    }

    private boolean filterMobalyticsLink(String word) {
        return word.contains("https://lor.mobalytics.gg/decks/code");
    }

    private String deckCodeFromMobaLinkUrl(String url) {
        return url.substring(url.lastIndexOf("/") + 1);
    }
}

File: backend/src/main/kotlin/info/gamewise/lor/videos/deckcodeextractor/DescriptionWithDeckLinkExtractor.kt
package info.gamewise.lor.videos.deckcodeextractor

import org.springframework.stereotype.Component

@Component
class DescriptionWithDeckLinkExtractor : AbstractDeckCodeExtractor(), DeckCodeExtractor {

    override fun extract(videoDescription: String): String? {
        val mobaUrl = descriptionAsStream(videoDescription)
            .firstOrNull { filterMobalyticsLink(it) }
        return deckCodeFromMobaLinkUrl(mobaUrl)
    }

    private fun filterMobalyticsLink(word: String): Boolean {
        return word.contains("https://lor.mobalytics.gg/decks/code")
    }

    private fun deckCodeFromMobaLinkUrl(url: String?): String? {
        return url?.substring(url.lastIndexOf("/") + 1)
    }
}